static/src/javascripts/projects/common/modules/navigation/user-account.ts

```diff
@@ -0,0 +1,141 @@
+				.then((commentLink) => {
+					if (commentLink) {
+						void fastdom.mutate(() => {
+							commentItem.classList.remove('u-h');
+							commentLink.setAttribute(
+								'href',
+								`https://profile.theguardian.com/user/id/${user.id}`,
+							);
+						});
+					}
+				});
+		});
+	}
+};
+
+const groupNotificationsByTarget = (
+	notifications: NotificationEvent[],
+): Record<string, string[]> => {
+	const notificationsMap: Record<string, string[]> = {};
+	notifications.forEach(({ message, target }) => {
+		if (Array.isArray(notificationsMap[target])) {
+			notificationsMap[target].push(message);
+		} else {
+			notificationsMap[target] = [message];
+		}
+	});
+	return notificationsMap;
+};
+
+const showNotifications = (notifications: NotificationEvent[]): void => {
+	void fastdom
+		.measure(() => ({
+			badge: document.querySelector(
+				'.js-user-account-notification-badge',
+			),
+			menu: document.querySelector('#my-account-dropdown'),
+		}))
+		.then((els) => {
+			const { badge, menu } = els;
+
+			if (notifications.length > 0 && badge && menu) {
+				// Show the notification badge
+				badge.classList.remove('is-hidden');
+
+				// Add messages to the relevant menu item(s)
+				const groupedNotifications =
+					groupNotificationsByTarget(notifications);
+
+				Object.entries(groupedNotifications).map(
+					([target, messages]) => {
+						const menuItem = menu.querySelector(
+							`a[data-link-id=${target}]`,
+						);
+						if (menuItem?.textContent) {
+							const labelEl = document.createElement('div');
+							labelEl.innerText = menuItem.textContent.trim();
+
+							const messageEls = messages.map((message) => {
+								const messageEl = document.createElement('div');
+								messageEl.classList.add(
+									'dropdown-menu__notification',
+								);
+								messageEl.innerText = message;
+								return messageEl;
+							});
+
+							menuItem.innerHTML = '';
+							labelEl.classList.add(
+								'top-bar__user-account-notification-badge',
+							);
+							[labelEl, ...messageEls].forEach((e) =>
+								menuItem.appendChild(e),
+							);
+						}
+					},
+				);
+			}
+		});
+};
+
+const showMyAccountIfNecessary = (): void => {
+	if (!isUserLoggedIn()) {
+		return;
+	}
+
+	void fastdom
+		.measure(() => ({
+			signIns: Array.from(
+				document.querySelectorAll('.js-navigation-sign-in'),
+			),
+			accountActionsLists: Array.from(
+				document.querySelectorAll('.js-navigation-account-actions'),
+			),
+			commentItems: Array.from(
+				document.querySelectorAll('.js-show-comment-activity'),
+			),
+		}))
+		.then((els) => {
+			const { signIns, accountActionsLists, commentItems } = els;
+			return fastdom
+				.mutate(() => {
+					signIns.forEach((signIn) => {
+						signIn.remove();
+					});
+					accountActionsLists.forEach((accountActions) => {
+						accountActions.classList.remove('is-hidden');
+					});
+
+					Array.from(
+						document.querySelectorAll('.js-user-account-trigger'),
```

Comment:
Should you selection not be part of the `fastdom.measure` step?

Reply:
possibly...this code is from the original .js and unrelated to our PR, so we'd rather not change things here!